src/Identity/UI/src/Areas/Identity/Pages/V3/Account/RegisterConfirmation.cshtml.cs

```diff
@@ -0,0 +1,90 @@
+    ///     directly from your code. This API may change or be removed in future releases.
+    /// </summary>
+    [AllowAnonymous]
+    [IdentityDefaultUI(typeof(RegisterConfirmationModel<>))]
+    public class RegisterConfirmationModel : PageModel
+    {
+        /// <summary>
+        ///     This API supports the ASP.NET Core Identity default UI infrastructure and is not intended to be used
+        ///     directly from your code. This API may change or be removed in future releases.
+        /// </summary>
+        public string Email { get; set; }
+
+        /// <summary>
+        ///     This API supports the ASP.NET Core Identity default UI infrastructure and is not intended to be used
+        ///     directly from your code. This API may change or be removed in future releases.
+        /// </summary>
+        public bool IsDevelopment { get; set; }
+
+        /// <summary>
+        ///     This API supports the ASP.NET Core Identity default UI infrastructure and is not intended to be used
+        ///     directly from your code. This API may change or be removed in future releases.
+        /// </summary>
+        public string EmailConfirmationUrl { get; set; }
+
+        /// <summary>
+        ///     This API supports the ASP.NET Core Identity default UI infrastructure and is not intended to be used
+        ///     directly from your code. This API may change or be removed in future releases.
+        /// </summary>
+        public virtual Task<IActionResult> OnGetAsync(string email) => throw new NotImplementedException();
+    }
+
+    internal class RegisterConfirmationModel<TUser> : RegisterConfirmationModel where TUser : class
+    {
+        private readonly UserManager<TUser> _userManager;
+        private readonly IWebHostEnvironment _hostingEnv;
+
+        public RegisterConfirmationModel(UserManager<TUser> userManager, IWebHostEnvironment hostingEnv)
+        {
+            _userManager = userManager;
+            _hostingEnv = hostingEnv;
+        }
+
+        public override async Task<IActionResult> OnGetAsync(string email)
+        {
+            if (email == null)
+            {
+                return RedirectToPage("/Index");
+            }
+
+            var user = await _userManager.FindByEmailAsync(email);
+            if (user == null)
+            {
+                return NotFound($"Unable to load user with email '{email}'.");
+            }
+
+            Email = email;
+            IsDevelopment = _hostingEnv.IsDevelopment();
+
+            if (IsDevelopment)
+            {
+                var userId = await _userManager.GetUserIdAsync(user);
+                var code = await _userManager.GenerateEmailConfirmationTokenAsync(user);
+                EmailConfirmationUrl = Url.Page(
+                    "/Account/ConfirmEmail",
+                    pageHandler: null,
+                    values: new { userId = userId, code = code },
+                    protocol: Request.Scheme);
+
```

Comment:
Extra line, but as mentioned above its better if you follow the `_ViewStart.cshtml` approach